cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -515,3 +569,26 @@ func removeTombstones(fn string) ([]*zoekt.Repository, error) {
 	}
 	return tombstones, nil
 }
+
+// deleteOrTombstone deletes the provided shards in indexDir that are associated with
+// the given repoID.
+//
+// If one of the provided shards is a compound shard and the repository is contained within it,
+// the repository is tombstoned instead.
+func deleteOrTombstone(indexDir string, repoID uint32, shardMerging bool, shards ...shard) {
```

Comment:
I see you mentioned you didn't like that we have to pass in indexDir and shardMerging to this func. Maybe instead it should be part of some helper struct that state. 

minor nit: I'd make `repoID` come after shardMerging param. IE the more "static" something is the closer to the front of the arg list. Just a habit from functional languages and currying.

Reply:
Got it...I'll play around with passing around some sort of helper to see if that's less awkward. Something like: 


```
type ServerConfig struct {
   shardMerging bool
   indexDir path

}

```